Automattic-Tracks-iOS/Event Logging/EventLoggingUploadManager.swift

```diff
@@ -0,0 +1,52 @@
+import Foundation
+import CocoaLumberjack
+import Sodium
+
+class EventLoggingUploadManager {
+
+    private struct Constants {
+        static let uuidHeaderKey = "log-uuid"
+        static let uploadHttpMethod = "POST"
+    }
+
+    internal var networkService = EventLoggingNetworkService()
+
+    internal var dataSource: EventLoggingDataSource!
+    internal var delegate: EventLoggingDelegate!
+
+    typealias LogUploadCallback = (Result<Void, Error>) -> Void
+
+    func upload(_ log: LogFile, then callback: @escaping LogUploadCallback) {
+
+        precondition(delegate != nil , "You must set the Event Logging Delegate prior to attempting an upload")
+        precondition(dataSource != nil , "You must set the Event Logging Data Source prior to attempting an upload")
+
+        guard delegate.shouldUploadLogFiles else {
+            delegate?.uploadCancelledByDelegate(log)
+            return
+        }
+
+        guard let fileContents = FileManager.default.contents(atUrl: log.url) else {
```

Comment:
I'd personally take `FileManager` as a constructor parameter as well. However, I understand that it's very common to use the `FileManager.default`.

Reply:
I reworked this to allow the developer to customize the instance of `FileManager` used, but putting it in the constructor seemed like a lot – it'd be exceedingly rare to use an alternative FileManager (I can't think of a file I've ever done it, and I couldn't find a single instance of it in WPiOS code – with the exception of one third-party Pod), so it would seem like clutter IMHO. 

Done in 4ea3255a864ec7f5b9cd22114162eddcbdb8fed9

Reply:
I kind of prefer the original over this 😬 I personally prefer to not have anything mutable unless it absolutely has to be. Like when the SDK forces you to make it mutable type of thing.

It looks like you made some changes to this since the commit, so I am just going to resolve this comment and start a new conversation about this if it's still necessary.